client/my-sites/marketplace/constants.ts

```diff
@@ -1,26 +1,123 @@
 /**
  * External dependencies
  */
-import { YOAST_SEO } from '@automattic/calypso-products';
+import { YOAST_PREMIUM, YOAST_FREE } from '@automattic/calypso-products';
 import debugFactory from 'debug';
 
 export const marketplaceDebugger = debugFactory( 'marketplace-debugger' );
 
 export const MARKETPLACE_FLOW_ID = 'marketplace_flow';
 export const ANALYTICS_UI_LOCATION_MARKETPLACE_DOMAIN_SELECTION = 'marketplace_domain_selection';
 
-// Marketplace plugin - product relationship mapped by SLUG
+/**
+ * Internal dependencies
+ */
+import {
+	IProductDefinition,
+	IProductCollection,
+	IProductGroupCollection,
+	YOAST,
+} from 'calypso/my-sites/marketplace/types';
+
+// ********** This section to be refactored out with new definition below **********
 export default interface PluginProductMappingInterface {
 	readonly 'wordpress-seo': string;
 	readonly 'wordpress-seo-premium': string;
 }
 
-// TODO: Refactor to a product indexed map of plugins
 export const PLUGIN_PRODUCT_MAP: PluginProductMappingInterface = {
-	'wordpress-seo': YOAST_SEO,
-	'wordpress-seo-premium': YOAST_SEO,
+	'wordpress-seo': YOAST_PREMIUM,
+	'wordpress-seo-premium': YOAST_PREMIUM,
 };
 
 export function getProductSlug( pluginSlug: keyof PluginProductMappingInterface ): string {
 	return PLUGIN_PRODUCT_MAP[ pluginSlug ];
 }
+// ************************************************************************************
+
+// TODO: Integrate this data structure with marketplace code
+export const productGroups: IProductGroupCollection = {
+	[ YOAST ]: {
+		products: {
+			[ YOAST_PREMIUM ]: {
+				defaultPluginSlug: 'wordpress-seo',
+				pluginsToBeInstalled: [ 'wordpress-seo-premium', 'wordpress-seo', 'yoast-test-helper' ],
+				isPurchasableProduct: true,
+			},
+			[ YOAST_FREE ]: {
+				defaultPluginSlug: 'wordpress-seo',
+				pluginsToBeInstalled: [ 'wordpress-seo' ],
+				isPurchasableProduct: false,
+			},
+		},
+	},
+};
+
+/**
+ * Provides the plugin that needs to be installed for a given product
+ *
+ * @param {string} productGroupSlug The product group the product belongs to
+ * @param {string} productSlug The product slug being setup.
+ * @returns {array[string]} An array of plugin slugs, returns null if product does slug not exist
+ */
+export function getPluginsToInstall(
+	productGroupSlug: keyof IProductGroupCollection,
+	productSlug: keyof IProductCollection
+): string[] | null {
+	const { pluginsToBeInstalled } = productGroups[ productGroupSlug ]?.products[ productSlug ] ?? {};
+	return pluginsToBeInstalled ? pluginsToBeInstalled : null;
```

Comment:
Since on L67 we default to an empty object, we'll never return a `null`. Did you mean to just put `null` instead of the `{}` that is used as a default value for the nullish coalescing, and then simply return `pluginsToBeInstalled`?

```suggestion
	const { pluginsToBeInstalled } = productGroups[ productGroupSlug ]?.products[ productSlug ] ?? null;
	return pluginsToBeInstalled;
```

Reply:
actually if we do this change when null is de-structured there will be an error right? 🤔

Reply:
Of course! Disregard.